Sources/SKCore/CompilationDatabaseBuildSystem.swift

```diff
@@ -100,42 +90,32 @@ extension CompilationDatabaseBuildSystem: BuildSystem {
   public func buildSettings(for document: DocumentURI, language: Language) async throws -> FileBuildSettings? {
     // FIXME: (async) Convert this to an async function once `CompilationDatabaseBuildSystem` is an actor.
     return await withCheckedContinuation { continuation in
-      self.queue.async {
-        continuation.resume(returning: self.settings(for: document))
-      }
+      continuation.resume(returning: self.settings(for: document))
     }
```

Comment:
Not sure if this gets cleaned up in a future commit, but just in case it isn't, the `withCheckedContinuation` could be removed here :)

Reply:
Also seems like the comment is now outdated

Reply:
Good catch. Thanks. I would have probably caught it sooner or later by fixing the TODO but we can do it now already.